mobile/lib/services/album.service.dart

```diff
@@ -441,4 +444,34 @@ class AlbumService {
       return false;
     }
   }
+
+  Future<Album?> getAlbumByName(String name, bool remoteOnly) async {
+    final query = _db.albums.filter().nameEqualTo(name).sharedEqualTo(false);
+
+    if (remoteOnly) {
+      return query.localIdIsNull().findFirst();
+    }
+
+    return query.findFirst();
```

Comment:
The query can be constructed in a  single statement using the Isar's [optional](https://isar.dev/queries.html#query-modifiers) method